app/Http/Controllers/GenreController.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Genre;
+use Exception;
+use Illuminate\Http\Request;
+
+class GenreController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     */
+    public function index()
+    {
+        try {
+            $genres = Genre::paginate(10);
```

Comment:
@damir-tahirovic You can leave 10 as a fallback (default) value, but perPage parameter should otherwise be read from the request parameters.